firebase-admin/runtime/src/main/java/io/quarkiverse/googlecloudservices/firebase/admin/runtime/FirebaseAdminProducer.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkiverse.googlecloudservices.firebase.admin.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.auth.FirebaseAuth;
+
+import io.quarkiverse.googlecloudservices.common.GcpBootstrapConfiguration;
+import io.quarkiverse.googlecloudservices.common.GcpConfigHolder;
+
+@ApplicationScoped
+public class FirebaseAdminProducer {
+
+    @Inject
+    GoogleCredentials googleCredentials;
+
+    @Inject
+    GcpConfigHolder gcpConfigHolder;
+
+    @Produces
+    @Singleton
+    @Default
+    public FirebaseAuth firestoreAuth() {
+        GcpBootstrapConfiguration gcpConfiguration = gcpConfigHolder.getBootstrapConfig();
+
+        FirebaseOptions.Builder firebaseOptionsBuilder = FirebaseOptions.builder()
+                .setCredentials(googleCredentials)
+                .setProjectId(gcpConfiguration.projectId.orElse(null));
+
+        return FirebaseAuth.getInstance(getFirebaseApp(gcpConfiguration, firebaseOptionsBuilder.build()));
+    }
+
+    private FirebaseApp getFirebaseApp(GcpBootstrapConfiguration gcpBootstrapConfiguration,
```

Comment:
I know this extension only support auth today but should it make sense to also provide FirebaseApp as a CDI bean ?

Reply:
This is a good idea. Until we don't add support for Realtime Database and Firebase Cloud Messaging, for example, provide a FirebaseApp bean will allow developers to create their own beans if needed.

I changed this piece of code providing also a FirebaseApp CDI bean.